cloudify_nsx/library/nsx_esg_dlr.py

```diff
@@ -1505,14 +1510,16 @@ def del_routing_rule(client_session, resource_id):
     common.check_raw_result(raw_result)
 
 
-def esg_dgw_clear(client_session, esg_id):
+def esg_dgw_clear(client_session, resource_id):
     """
     This function clears the default gateway config on an ESG
     :param client_session: An instance of an NsxClient Session
-    :param esg_id: The id of the ESG to list interfaces of
+    :param resource_id: response from esg_dgw_set
     :return: True on success, False on failure
     """
 
+    esg_id, _ = resource_id.split("|")
```

Comment:
You might need to limit the amount of splits you get back from this- e.g.
 `>>> test = 'say|hello|wave|goodbye'`
 `>>> x, _ = test.split('|')`
` Traceback (most recent call last):`
`  File "<stdin>", line 1, in <module>`
` ValueError: too many values to unpack`
` >>> x, _ = test.split('|', 1)`

(just in case the resource ID contains more elements at some point?)